api/context/src/main/java/io/opentelemetry/context/StrictContextStorage.java

```diff
@@ -247,67 +206,9 @@ public String toString() {
     final long threadId = currentThread().getId();
     final Context context;
 
-    volatile boolean closed;
-
     CallerStackTrace(Context context) {
       super("Thread [" + currentThread().getName() + "] opened scope for " + context + " here:");
       this.context = context;
     }
   }
-
-  static class PendingScopes extends WeakConcurrentMap<Scope, CallerStackTrace> {
-
-    static PendingScopes create() {
-      return new PendingScopes(new ConcurrentHashMap<>());
-    }
-
-    // We need to explicitly pass a map to the constructor because we otherwise cannot remove from
-    // it. https://github.com/raphw/weak-lock-free/pull/12
-    private final ConcurrentHashMap<WeakKey<Scope>, CallerStackTrace> map;
-
-    @SuppressWarnings("ThreadPriorityCheck")
-    PendingScopes(ConcurrentHashMap<WeakKey<Scope>, CallerStackTrace> map) {
-      super(/* cleanerThread= */ false, /* reuseKeys= */ false, map);
-      this.map = map;
-      // Start cleaner thread ourselves to make sure it runs after initializing our fields.
-      Thread thread = new Thread(this);
-      thread.setName("weak-ref-cleaner-strictcontextstorage");
-      thread.setPriority(Thread.MIN_PRIORITY);
-      thread.setDaemon(true);
-      thread.start();
-    }
-
-    List<CallerStackTrace> drainPendingCallers() {
-      List<CallerStackTrace> pendingCallers =
-          map.values().stream().filter(caller -> !caller.closed).collect(Collectors.toList());
-      map.clear();
-      return pendingCallers;
-    }
-
-    // Called by cleaner thread.
```

Comment:
This is removing the functionality of the `StrictContextStorage` completely, which would be caught if the test wasn't deleted with it. 

Anyways we will need to figure out how to allow instrumentation to call this from tests as well and we don't need to delete this code.

Reply:
I am lost in whole this code. How can it happen to have this check without getting an error in one of the other places:
https://github.com/open-telemetry/opentelemetry-java/blob/main/api/context/src/main/java/io/opentelemetry/context/ThreadLocalContextStorage.java#L34

OR 

https://github.com/open-telemetry/opentelemetry-java/blob/main/api/context/src/main/java/io/opentelemetry/context/StrictContextStorage.java#L226

Reply:
The test for garbage leaked scopes is only in one place

https://github.com/open-telemetry/opentelemetry-java/blob/main/api/context/src/strictContextEnabledTest/java/io/opentelemetry/context/StrictContextStorageTest.java#L170

Reply:
If the error pattern is to always start a new scope and never close we can add a check for the scope depth and log error when larger than 1000 or something like that

Reply:
I think that assumes the user doesn't try to work around context issues with `Context.root().makeCurrent()`, which java instrumentation currently does since we have scope bugs and don't use this debug mechanism yet. Checking for GC-leaked scopes is more direct with less corner cases I think.

Reply:
I will have to think about that. But we should keep this discussion open and if we are able to get rid of the GC part would be great. We can have the GC logic as a test helper for testing purposes, but if the logic controlled by this flag gets simpler and users can run with this with a reasonable overhead may actually be better for them. But let's brainstorm here

Reply:
Sure - for reference, the majority of the overhead of the flag is from recording the stacktrace when attaching, which makes finding a leak much easier than only knowing the stack when detaching. The GC listener is something, but very minor in comparison to that.